internal/layer2/announcer.go

```diff
@@ -98,7 +111,7 @@ func (a *Announce) updateInterfaces() {
 			if !ok {
 				continue
 			}
-			if ipaddr.IP.To4() != nil || !ipaddr.IP.IsLinkLocalUnicast() {
+			if ipaddr.IP.To4() != nil || ipaddr.IP.IsLinkLocalUnicast() {
```

Comment:
I am not sure this is correct, plus v6 has been working without issues and if this wasn't right we'd have issues.

Reply:
In fact, We hit this issue. This is because the ipv6 linkLocalUnicast address. Refer to [dadfailed](https://blog.tankywoo.com/2013/09/27/ipv6-dadfailed-problem.html).

I knew https://github.com/metallb/metallb/pull/1726 can fixed this issue, But in fact, we should really ignore the ipv6 local link address, right?

Reply:
I need to read a bit more. Can you remove this from this pr so we can move forward with the rest? It's not fitting anyway.

Reply:
sure, The changes here should be reflected in another PR, so I'll start a new one to address this.

Reply:
see https://github.com/metallb/metallb/pull/1809